daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetImpl.scala

```diff
@@ -95,6 +96,10 @@ sealed trait DINode {
 
 }
 
+object DINode{
+  implicit def toDINode(v: DataValueDINodeTrait):DINode = v.asInstanceOf[DINode]
```

Comment:
This cast is only needed because of the daffodil-lib location of the DataValue class. This whole thing can be mixins to DINode and then there's no casting I think.

Reply:
To need for a mixin. This whole exercise is just because DataValue doesn't have access to the DINode type.